### PR TITLE
Patch `Ingredient#isEmpty` for custom ingredients

### LINKED ISSUE #1726: Error in log file :Recipe minecraft:*_chest_boat can't be placed due to empty ingredients and will be ignored
**Minecraft Version:** 1.21.4

**Logs:**:
```
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:acacia_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:bamboo_chest_raft can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:birch_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:cherry_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:chest_minecart can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:dark_oak_chest_boat can't be placed due to empty ingredients and will be ignore
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:hopper can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:jungle_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:mangrove_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:oak_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:pale_oak_chest_boat can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:shulker_box can't be placed due to empty ingredients and will be ignored
[17:14:26] [Render thread/WARN] [minecraft/RecipeManager]: Recipe minecraft:spruce_chest_boat can't be placed due to empty ingredients and will be ignored
```

**Steps to Reproduce:**
Start the game.
Generate a world

**Description of issue:**
Strange errors in the log file

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Seems like we forgot to patch this new method to use the custom ingredient item list.
Fixes #1726

### CODE REVIEW

**Review by Technici4n — DISMISSED** [review]

**Review by embeddedt — CHANGES_REQUESTED** [review]
I don't like the duplicate population of `this.customIngredientValues`; IMO that should be extracted to a common private helper method.

**Review by embeddedt — DISMISSED** [review]

File: patches/net/minecraft/world/item/crafting/Ingredient.java.patch [hunk 1]
```diff
@@ -124,6 +118,12 @@
 +
 +    public boolean isCustom() {
 +        return this.customIngredient != null;
++    }
++
++    private void updateCustomIngredientValues() {
```
Comment (Technici4n) [inline patches/net/minecraft/world/item/crafting/Ingredient.java.patch]: Just make it a getter that returns the list, no?

**Review by embeddedt — APPROVED** [review]

**Review by Technici4n — CHANGES_REQUESTED** [review]
Not a great method name but I don't care. :D

**Review by Technici4n — APPROVED** [review]
Sorry I meant to approve.

**Pull request merged**